Key Rotation (#284)
ietf-wg-scitt/draft-ietf-scitt-architecture: draft-ietf-scitt-architecture.md

The Transparency Service may be replicated with a consensus algorithm, such as Practical Byzantine Fault Tolerance {{PBFT}} and may be used to protect against malicious or vulnerable replicas.
Threshold signatures may be use to protect the service key, etc.

> Issuers and Transparency Services MUST their rotate keys, see {{KEY-MANAGEMENT}}.

Reviewer: ```suggestion
Issuers and Transparency Services MUST rotate their keys, see {{KEY-MANAGEMENT}}.
```

> Issuers and Transparency Services MUST rotate their keys, see {{KEY-MANAGEMENT}}.

Reviewer 2: NIST Special Publication 800-57 Part 3
Cryptoperiod is defined in 

Revision 1
Recommendation for
Key Management
Part 3: Application-Specific Key 
Management Guidance

> Issuers and Transparency Services MUST rotate their keys, see {{KEY-MANAGEMENT}}.

Author: ```suggestion
Issuers and Transparency Services MUST rotate their keys in well-defined cryptoperiods, see {{KEY-MANAGEMENT}}.
```

> Issuers and Transparency Services MUST rotate their keys in well-defined cryptoperiods, see {{KEY-MANAGEMENT}}.

Author: ```suggestion
Issuers and Transparency Services MUST rotate their keys in well-defined crypto-periods, see {{KEY-MANAGEMENT}}.
```

> Issuers and Transparency Services MUST rotate their keys in well-defined cryptoperiods, see {{KEY-MANAGEMENT}}.

Reviewer 3: Might be better to make this a SHOULD, sometimes keys cannot be rotated.

Reviewer 4: With this in the security section, not in the normative, I'm ok with this change as it doesn't change interop, but rather a quality judgment of a TS that doesn't rotate keys.
```suggestion
Issuers and Transparency Services SHOULD rotate their keys in well-defined cryptoperiods, see {{KEY-MANAGEMENT}}.
```

Reviewer 5: I'd be happy to approve with this change integrated.

Reviewer 6: Same here, I would merge this as it is:

I am uncomfortable diluting MUST to SHOULD. Our Architecture, MUST require this at the base.

in case, if any profile cannot implement this, they can state that in their implementation!

Reviewer 2: Orie's posits that there is a situation where keys can never be rotated, but is that a practical position?

Reviewer 3: An example of a key that cannot be rotated is a burned in key that attests to things. @henkbirkholz 

When SHOULD is used, there needs to be an explanation of when the advice can be ignored, and when it is MUST.

When possible, keys MUST be rotated, unless it is not possible to attribute new keys to the same issuer identity, etc...

> Issuers and Transparency Services MUST rotate their keys in well-defined crypto-periods, see {{KEY-MANAGEMENT}}.

Reviewer 7: ```suggestion
Issuers and Transparency Services MUST rotate their keys in well-defined cryptoperiods, see {{KEY-MANAGEMENT}}.
```
